jsapp/js/components/modalForms/translationSettings.es6

```diff
@@ -0,0 +1,324 @@
+    let translated = content.translated,
+        translationsLength = content.translations.length,
+        survey = content.survey,
+        choices = content.choices;
+
+    // append null values to translations for each survey row
+    for (var i = 0, len = survey.length; i < len; i++) {
+      let row = survey[i];
+      for (var j = 0, len2 = translated.length; j < len2; j++) {
+        var property = translated[j];
+        if (row[property] && row[property].length < translationsLength) {
+          row[property].push(null);
+        }
+      }
+    }
+
+    // append null values to translations for choices
+    if (content.choices && content.choices.length) {
+      for (var i = 0, len = choices.length; i < len; i++) {
+        if (choices[i].label.length < translationsLength) {
+          choices[i].label.push(null);
+        }
+      }
+    }
+    return content;
+  }
+  deleteTranslations(content, langIndex) {
+    let translated = content.translated,
+        translationsLength = content.translations.length,
+        survey = content.survey,
+        choices = content.choices;
+
+    for (var i = 0, len = survey.length; i < len; i++) {
+      let row = survey[i];
+      for (var j = 0, len2 = translated.length; j < len2; j++) {
+        var property = translated[j];
+        if (row[property]) {
+          if (row[property].length === translationsLength) {
+            row[property].splice(langIndex, 1);
+          } else {
+            console.error('Translations index mismatch');
+            return false;
+          }
+        }
+      }
+    }
+
+    for (var i = 0, len = choices.length; i < len; i++) {
+      if (choices[i].label) {
+        if (choices[i].label.length === translationsLength) {
+          choices[i].label.splice(langIndex, 1);
+        } else {
+          console.error('Translations index mismatch');
+          return false;
+        }
+      }
+    }
+    return content;
+  }
+  updateAsset (content) {
+    actions.resources.updateAsset(
+      this.props.asset.uid,
+      {content: JSON.stringify(content)}
+    );
+  }
+  render () {
+    let translations = this.state.translations;
+    return (
+      <bem.FormModal m='translation-settings'>
+        <bem.FormModal__item>
+          {(translations && translations[0] === null) ?
+            <bem.FormView__cell m='translation-note'>
+              {t('Here you can add one more more languages to your project, and translate the strings in each language.')}
+              &nbsp;
+              <em>
+                {t('Note: make sure your default language has a name. If it doesn\'t, you will not be able to edit your form in the form builder.')}
+              </em>
+            </bem.FormView__cell>
+            :
+            <bem.FormView__cell m='label'>
+              {t('Current languages')}
+            </bem.FormView__cell>
+          }
+          {translations.map((l, i)=> {
+            return (
+              <React.Fragment key={`lang-${i}`}>
```

Comment:
I didn't know about this, up until now I was using additional `<div>`s to wrap up multiple children 👎